### ISSUE TITLE
StoredPaymentMethodDetails doesn't have support for the "subtype" field used by the native TWINT drop-in

### ISSUE THREAD

**Issue by elitalon** [Adyen/adyen-java-api-library#1360]
**Describe the bug**
Deserialisation of a [`POST /payments`](https://docs.adyen.com/api-explorer/Checkout/latest/post/payments#request-paymentMethod-StoredPaymentMethodDetails) request done against a proxy backend ignores the `subtype` field from the `paymentMethod` object in the JSON body.

This results in the drop-in integration of TWINT for iOS using a web-based UI instead of the native one.

**How to reproduce**
Make a [`POST /payments`](https://docs.adyen.com/api-explorer/Checkout/latest/post/payments) request with the following JSON body:

```json
{
    "channel": "iOS",
    "amount": {
        "currency": "CHF",
        "value": 0
    },
    "browserInfo": {
        "userAgent": "<redacted>"
    },
    "merchantAccount": "<redacted>",
    "returnUrl": "<redacted>",
    "storePaymentMethod": true,
    "countryCode": "CH",
    "paymentMethod": {
        "type": "twint",
        "subtype": "sdk"
    },
    "shopperLocale": "en_US",
    "shopperReference": "<redacted>",
    "reference": "<redacted>",
    "recurringProcessingModel": "<redacted>",
    "shopperInteraction": "Ecommerce"
}
```

**Expected behavior**
The resulting [`StoredPaymentMethodDetails`](https://github.com/Adyen/adyen-java-api-library/blob/9bb424b6b40d82a7af7d9db8b7edef7d1ba75e7d/src/main/java/com/adyen/model/checkout/StoredPaymentMethodDetails.java) object stores the value of the `subtype` JSON field.

**Screenshots**
Not applicable

**Desktop (please complete the following information):**
 - Java Version: Java 21
 - Library Version: [28.2.0](https://github.com/Adyen/adyen-java-api-library/releases/tag/v28.2.0)

**Additional context**
This problem was discovered as part of https://github.com/Adyen/adyen-ios/issues/1783.

**Comment by jillingk** [comment]
Hi @elitalon,

Thanks for reaching out here, we have a fix on the way!

Best, Jilling
Adyen

**Comment by elitalon** [comment]
Hi @jillingk.

Are there any news on this? I've seen a few releases of the library in the past weeks, but none of them seem to mention this issue.

Thanks!

**Comment by DjoykeAbyah** [comment]
Hi @elitalon!

You can expect the fix in the upcoming release in 2 weeks. See pull request #1380 for progress. [src/main/java/com/adyen/model/checkout/TwintDetails.java](https://github.com/Adyen/adyen-java-api-library/pull/1380/files#diff-6b679667f19e0dd924e65fb71c21f1886e58a87886a609ad94a30896d7ccff2d). 

Kind regards,

Djoyke
Adyen

**Comment by DjoykeAbyah** [comment]
Hi @elitalon!

We've merged the fix from this PR (#1380): [src/main/java/com/adyen/model/checkout/TwintDetails.java](https://github.com/Adyen/adyen-java-api-library/pull/1380/files#diff-6b679667f19e0dd924e65fb71c21f1886e58a87886a609ad94a30896d7ccff2d).

Would you be willing to check the fix and confirm that it addresses the issue before we proceed with the release?

We appreciate your help!

Best regards,
Djoyke
Adyen

**Comment by elitalon** [comment]
@DjoykeAbyah Thanks for the heads up 👍🏼 I will give it a try.

**Comment by elitalon** [comment]
@DjoykeAbyah Speaking of testing, how do you intend us to reach that change without [a new version](https://github.com/Adyen/adyen-java-api-library/pull/1383)?

Specifically, how can we point to the current state of the `main` branch in the `pom.xml` of our project?

**Comment by gcatanese** [comment]
Hey @elitalon thanks for helping out. We are not able to deploy a snapshot yet (we are looking at the options and we hope to do so in the near future) so it is indeed difficult to test.
The only option at the moment would be to build the library from source (`main` branch).

If that's not doable I understand, if you can please check the changes introduced by the PR #1380 and let us know if they look what you expect, that would be tremendously helpful. We are planning the new release shortly.

**Comment by elitalon** [comment]
@gcatanese I'll try building from source. If that doesn't work, then I'll take a look at the pull request. Cheers!

**Comment by elitalon** [comment]
Hi @gcatanese, I was able to build from source and test the integration agains our backend. I also took a look at #1380.

I think we should be good to go, as far as this issue is concerned.

It could be that we find something else when trying an end-to-end registration together with the iOS SDK. But in that case we will open a separate issue.

Thanks a lot!

**Comment by gcatanese** [comment]
That's great, thank you so much @elitalon for your help.

**Comment by DjoykeAbyah** [comment]
Hi @elitalon,

I wanted to quickly mention the Java library has just been released with the changes from #1380!

Kind Regards,

Djoyke
Adyen

**Comment by elitalon** [comment]
Thanks a lot, @DjoykeAbyah!
